src/zarr/codecs/sharding.py

```diff
@@ -631,6 +621,7 @@ def _get_index_chunk_spec(self, chunks_per_shard: ChunkCoords) -> ArraySpec:
             shape=chunks_per_shard + (2,),
             dtype=np.dtype("<u8"),
             fill_value=MAX_UINT_64,
+            order="C",  # TODO!!!!
```

Comment:
```suggestion
            order="C",
```

A hard-coded `C` is fine here, because it is just the shard index. It doesn't surface into user code.